sources/packages/api/src/route-controllers/institution-locations/models/institution-location.dto.ts

```diff
@@ -1,48 +1,81 @@
-import { InstitutionPrimaryContact } from "../../../types";
+import { Allow } from "class-validator";
 import { InstitutionLocationInfo } from "../../../database/entities/institution-location.model";
+import { AddressInfoOutDTO } from "../../models/common.dto";
 
 export class GetInstitutionLocationDto {
   id: number;
   data: InstitutionLocationInfo;
   name: string;
 }
 /**
- * Interface which defines the contract of how institution location data is passed to form.io
+ * Payload/Input DTO for institution location.
+ ** Class validators are not used for DTO validation
+ ** because of dry-run validation.
  */
-export interface InstitutionLocationTypeDto {
+export class InstitutionLocationFormAPIInDTO {
+  @Allow()
   addressLine1: string;
+  @Allow()
   addressLine2?: string;
+  @Allow()
   city: string;
+  @Allow()
   country: string;
+  @Allow()
   locationName: string;
+  @Allow()
   postalCode: string;
+  @Allow()
   provinceState: string;
+  @Allow()
   institutionCode: string;
+  @Allow()
   primaryContactFirstName: string;
+  @Allow()
   primaryContactLastName: string;
+  @Allow()
   primaryContactEmail: string;
+  @Allow()
   primaryContactPhone: string;
 }
 
 /**
- * Interface which defines the contract of how institution location data is passed to Vue component
+ * Response/Output DTO for institution location.
  */
-export class InstitutionLocationsDetailsDto {
+export class InstitutionLocationFormAPIOutDTO {
+  addressLine1: string;
+  addressLine2?: string;
+  city: string;
+  country: string;
+  locationName: string;
+  postalCode: string;
+  provinceState: string;
+  institutionCode: string;
+  primaryContactFirstName: string;
+  primaryContactLastName: string;
+  primaryContactEmail: string;
+  primaryContactPhone: string;
+}
+
+export class InstitutionPrimaryContactDTO {
+  primaryContactFirstName: string;
+  primaryContactLastName: string;
+  primaryContactEmail: string;
+  primaryContactPhone: string;
+}
+
+/**
+ * DTO which defines the contract of how institution location data is passed to Vue component
+ */
+export class InstitutionLocationAPIOutDTO {
   id: number;
   name: string;
   data: {
-    address: {
-      addressLine1: string;
-      addressLine2?: string;
-      province: string;
-      country: string;
-      city: string;
-      postalCode: string;
-    };
+    address: AddressInfoOutDTO;
   };
-  primaryContact: InstitutionPrimaryContact;
```

Comment:
mm, why do we need to add DTO to this interface? I see that this is a generic interface, which is also used in models, 
![image](https://user-images.githubusercontent.com/77353155/163042619-93158855-797d-4c09-93c8-440ef91b0c87.png)

Reply:
But, when used by a parent DTO it may need class validators.  So making DTO to keep that in API context.

Reply:
If we need changes at the API level we definitely need to create the DTO, that said, when changes are not needed we usually are not recreating the classes at the other levels.
For instance, if we need a class instead of an interface at the API level in order to Swagger generate the proper documentation, maybe it would be a reason to have it at the API level.